File: exercise.js
class Vec {
  constructor(x, y) {
    this.x = x;
    this.y = y;
  }

  plus(vec) {
    return new Vec((this.x + vec.x), (this.y + vec.y));
  }

  minus(vec) {
    return new Vec((this.x - vec.x), (this.y - vec.y));
  }

  get length() {
    return Math.sqrt(Math.pow(this.x, 2) + Math.pow(this.y, 2));
  }

  static distance(point1, point2) {
    return Math.sqrt((Math.pow(point2[0] - point1[0], 2)) + (Math.pow(point2[1] - point1[1])));
  }
}

File: exercise.py
import math

class Vec:
    def __init__(self, x, y):
        self.x = x
        self.y = y
        self.__length = math.sqrt(self.x ** 2, self.y ** 2)

    def plus(self, new_vec):
        return Vec((self.x + new_vec.x), (self.y + new_vec.y))

    def minus(self, new_vec):
        return Vec((self.x - new_vec.x), (self.y - new_vec.y))

    @property
    def length(self):
        return self.__length

    @staticmethod
    def distance(self, point1, point2):
        return math.sqrt((point2[0] - point1[0])**2 + (point2[1] - point1[1])**2)